code/xcvm/lib/core/src/gateway.rs

```diff
@@ -49,17 +59,47 @@ pub enum ExecuteMsg {
 	Bridge(BridgeMsg),
 
 	/// Message sent by an admin to register a new asset.
-	RegisterAsset {
-		asset_id: AssetId,
-		reference: AssetReference,
-	},
+	RegisterAsset(RegisterAssetMsg),
 
 	/// Message sent by an admin to remove an asset from registry.
 	UnregisterAsset {
 		asset_id: AssetId,
 	},
+
+	Wasm(Ics20MessageHook),
+}
+
+/// when message is sent to other side, we should identify receiver of some kind
+#[derive(Serialize, Deserialize, JsonSchema, Clone, PartialEq, Debug)]
```

Comment:
```suggestion
#[cw_serde]
#[derive(Debug)]
```

Reply:
i cannot do it. because will not be able to compile for substrate. this lib is for cross consumption of these things

Reply:
Sorry, didn’t notice it was in lib/core.